main/field.h

```diff
@@ -79,6 +79,9 @@ typedef enum eFieldDataType {
 	FIELDTYPE_END_MARKER = 1 << 3,
 } fieldDataType;
 
+#define FIELD_NULL_LETTER_CHAR '-'
+#define FIELD_NULL_LETTER_STRING "-"
```

Comment:
field.h is visible to parsers.

fmt.c is a part of the `main` part. So let's put the definition to field_p.h.
The _p.h files are private to the `main` part.